### PR TITLE
Fix #1464

### LINKED ISSUE #1464: Assertion 'httpparser_type == HTTP_REQUEST || httpparser_type == HTTP_RESPONSE' failed
###### IoT.js version:

```
Checked revision: ab10ae5
Build command: tools/build.py --buildtype debug --jerry-profile=es2015-subset --experimental
```
###### OS:

```
Ubuntu 17.10, x86_64
```
###### Test case:

```javascript
var http_common = require('http_common');
http_common.createHTTPParser().reinitialize(-7581);
```

###### Backtrace:

```
iotjs/src/modules/iotjs_module_httpparser.c:380: Assertion 'httpparser_type == HTTP_REQUEST || httpparser_type == HTTP_RESPONSE' failed.
```

<sup>Found by [Fuzzinator](http://fuzzinator.readthedocs.io/) with [grammarinator](https://github.com/renatahodovan/grammarinator)</sup>

### PR DESCRIPTION
**Pull request by daeyeon**
IoT.js-DCO-1.0-Signed-off-by: Daeyeon Jeong daeyeon.jeong@samsung.com

### CODE REVIEW

File: src/modules/iotjs_module_httpparser.c [hunk 1]
```diff
@@ -376,8 +376,10 @@ JS_FUNCTION(Reinitialize) {
   DJS_CHECK_ARGS(1, number);
 
   http_parser_type httpparser_type = (http_parser_type)(JS_GET_ARG(0, number));
-  IOTJS_ASSERT(httpparser_type == HTTP_REQUEST ||
-               httpparser_type == HTTP_RESPONSE);
+
+  if (httpparser_type != HTTP_REQUEST || httpparser_type != HTTP_RESPONSE) {
```
Comment (hs0225) [inline src/modules/iotjs_module_httpparser.c]: `httpparser_type != HTTP_REQUEST && httpparser_type != HTTP_RESPONSE`

File: src/modules/iotjs_module_httpparser.c [hunk 2]
```diff
@@ -454,8 +456,11 @@ JS_FUNCTION(HTTPParserCons) {
   const jerry_value_t jparser = JS_GET_THIS();
 
   http_parser_type httpparser_type = (http_parser_type)(JS_GET_ARG(0, number));
-  IOTJS_ASSERT(httpparser_type == HTTP_REQUEST ||
-               httpparser_type == HTTP_RESPONSE);
+
+  if (httpparser_type != HTTP_REQUEST || httpparser_type != HTTP_RESPONSE) {
```
Comment (hs0225) [inline src/modules/iotjs_module_httpparser.c]: ditto.

File: src/modules/iotjs_module_httpparser.c (hunk 1 above)
Comment (daeyeon) [reply src/modules/iotjs_module_httpparser.c]: Thanks. they're fixed.

Comment (daeyeon) [reply src/modules/iotjs_module_httpparser.c]: Thanks. Fixed.

**Pull request merged**